libs/shared/src/lib/components/extract/index.tsx

```diff
@@ -2,23 +2,55 @@
 import { BytebankText } from '../text';
 import { Box, Card } from '@mui/material';
 import { format } from 'date-fns';
+import { useEffect, useState } from 'react';
+import { useUser } from '../../contexts/user.context';
+import {
+  BytebankExtractProps,
+  ExtractJsonProps,
+} from '../../classes/models/extract';
 
-export interface BytebankExtractProps {
-  month: string;
-  data: BytebankExtractPropsData[];
-}
+export function BytebankExtract() {
+  const [extract, setExtract] = useState<BytebankExtractProps[]>([]);
+  const { user } = useUser();
 
-interface BytebankExtractPropsData {
-  date: Date;
-  type: 'Depósito' | 'Saque' | 'Transferência';
-  value: number;
-}
+  const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+
+  const fetchExtract = async () => {
+    const res = await fetch(`${apiUrl}/extract?userId=${user?.id}`);
+    const extract = await res.json();
+    return extract as ExtractJsonProps[];
+  };
+
+  useEffect(() => {
+    fetchExtract().then((res) => {
+      console.log(res, user);
+      if (res.length === 0) {
+        return;
+      }
+      const resData = res[0].transactions;
+      const agrupado: BytebankExtractProps[] = Object.values(
+        resData.reduce((acc, item) => {
+          const dataObj = new Date(item.date);
+          const mes = format(dataObj, 'MMMM');
+
+          if (!acc[mes]) {
+            acc[mes] = { month: mes, data: [] };
+          }
+
+          acc[mes].data.push({
+            date: dataObj,
+            type: item.type,
+            value: item.value,
+          });
+
+          return acc;
+        }, {} as Record<string, BytebankExtractProps>)
+      );
+
+      setExtract(agrupado);
+    });
+  }, [user]);
 
-export function BytebankExtract({
-  extract,
-}: {
-  extract: BytebankExtractProps[];
-}) {
   const numeroFormatado = (number: number) =>
```

Comment:
Sugestão para manter o padrão do nome de métodos: 
```suggestion
  const handleFormatValue = (number: number) =>
```